packages/arb-avm-cpp/cavm/ccheckpointstorage.cpp

```diff
@@ -229,3 +229,26 @@ int deleteData(CCheckpointStorage* storage_ptr,
     auto status = keyvalue_store->deleteData(key_vector);
     return status.ok();
 }
+
+ByteSliceArray getKeysWithPrefix(CCheckpointStorage* storage_ptr,
```

Comment:
Looking back at this and seeing the complexity of the checkpointer, my view is currently that providing more specialized support in the backend is worth it since the checkpointing logic can be made way simpler. This is still a WIP so I haven't yet written up the design in the top of this PR and I'm still iterating on it.